### PR TITLE
Plat 278 update pending counters in incoming requests

### PR DESCRIPTION
**Pull request by MikhailZhukov**
**- What I did**
Increment pending counters when processing new message

**- How I did it**
Add increment code to SMExecute.stepWaitObjectReady

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/v2/conveyor/smachine/api_shared_link.go [hunk 1]
```diff
@@ -8,10 +8,23 @@ package smachine
 import (
 	"fmt"
 	"reflect"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
 )
 
 type SharedDataFunc func(interface{}) (wakeup bool)
 
+func NewUnboundSharedData(data interface{}) SharedDataLink {
```
Comment (ivanshibitov) [inline ledger-core/v2/conveyor/smachine/api_shared_link.go:17]: What for?

Comment (ivanshibitov) [reply ledger-core/v2/conveyor/smachine/api_shared_link.go:17]: I mean this function is not used anywhere

Comment (MikhailZhukov) [reply ledger-core/v2/conveyor/smachine/api_shared_link.go:17]: This function create SharedDataLink which sometimes need for tests. But unit test was deleted (it does not work). Do i must remove this function too?

Comment (MikhailZhukov) [reply ledger-core/v2/conveyor/smachine/api_shared_link.go:17]: Yes, it does not used right now. I can remove it. But maybe it useful for future tests?

Comment (MikhailZhukov) [reply ledger-core/v2/conveyor/smachine/api_shared_link.go]: remove this code

File: ledger-core/v2/virtual/execute/execute.go [hunk 2]
```diff
@@ -145,6 +145,12 @@ func (s *SMExecute) stepWaitObjectReady(ctx smachine.ExecutionContext) smachine.
 		semaphoreOrdered = state.MutableExecute
 		semaphoreUnordered = state.ImmutableExecute
 
+		if isOrdered {
```
Comment (ivanshibitov) [inline ledger-core/v2/virtual/execute/execute.go:148]: Lower, There is following code
if ctx.AcquireForThisStep(semaphoreReadyToWork).IsNotPassed() {
		return ctx.Sleep().ThenRepeat()
}
it means that if we can't acquire the lock we will repeat this step and again increment this numbers

Comment (MikhailZhukov) [reply ledger-core/v2/virtual/execute/execute.go]: Fix multiple increment on retries

File: ledger-core/v2/virtual/object/object.go [hunk 3]
```diff
@@ -30,10 +30,23 @@ type Info struct {
...
-	PotentialMutablePendingCount   uint8
+	ActiveImmutablePendingCount           uint8
+	ActiveMutablePendingCount             uint8
+	PotentialImmutablePendingCount        uint8
+	PotentialMutablePendingCount          uint8
+	potentialPendingCounterWasIncremented bool
+}
+
+func (i *Info) IncrementPotentialPendingCounterOnce(isOrdered bool) {
+	if i.potentialPendingCounterWasIncremented {
```
Comment (bigbes) [inline ledger-core/v2/virtual/object/object.go:41]: I don't get that logic, so we can increment only once?

**Pull request merged**